### PR TITLE
Fix: env0_variable_set_assignment 404, Error: failed to unassign vari…

### LINKED ISSUE #966: `env0_variable_set_assignment` 404, Error: failed to unassign variable sets: 404 Not Found: Configuration Set "..." is not assigned to project.
**Describe the bug**
When using `env0_variable_set_assignment` on a project, and another variable set gets assigned to the same project (manually); and the subsequent plan causes a 404 error.

**To Reproduce**
1. create three org level variable sets
2. assign one org variable set to a project (using provider), assign one org variable set to a child project (using provider)
3. assign second org variable set to same child project (by hand)
4. re-run provider code from step 2
5. see plan - suggesting removal of variable set added in step 3
6. approve plan - get 404 error for an unrelated variable set.

**Expected behavior**
No error; removal of hand set variable set.

**Provider Version**
1.20.12

**Screenshots**

```
OpenTofu will perform the following actions:

  # env0_variable_set_assignment.child will be updated in-place
  ~ resource "env0_variable_set_assignment" "child" {
        id       = "ea74945b-f52a-491f-a88e-ec93e7941244"
      ~ set_ids  = [
            "1d199243-455d-4af8-be22-aed5401d5016",
          - "f04e33ac-1494-4ae3-a008-71497251acf4",
        ]
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
env0_variable_set_assignment.child: Modifying... [id=ea74945b-f52a-491f-a88e-ec93e7941244]
╷
│ Error: failed to unassign variable sets: 404 Not Found: Configuration Set "9f3048b3-6b12-4d28-904c-f0fa2fb2250c" is not assigned to project (ea74945b-f52a-491f-a88e-ec93e7941244)
│
│   with env0_variable_set_assignment.child,
│   on main.tf line 46, in resource "env0_variable_set_assignment" "child":
│   46: resource "env0_variable_set_assignment" "child" {
│
╵
```

note: set `f04e33...` was assigned manually.
set `9f3048...` was assigned in the parent project (and inherited into the child project)
project id: `ea74945b-f52a-491f-a88e-ec93e7941244` is the child project
variable sets are all created at the project level.

**Additional context**

See sample code:

```
terraform {
  required_providers {
    env0 = {
      source  = "env0/env0"
      version = ">=1.20.12"
    }
  }
}

provider "env0" {
  # Configuration options
}


data "env0_variable_set" "test1" {
  name  = "test_org_var_set"
  scope = "ORGANIZATION"
}

data "env0_variable_set" "test2" {
  name  = "test_org_var_set2"
  scope = "ORGANIZATION"
}

variable "project_id" {
  type    = string
  default = "37a67127-db31-4fce-9e15-b9df6e7f8aea"
}

variable "child_project_id" {
  type    = string
  default = "ea74945b-f52a-491f-a88e-ec93e7941244"
}

resource "env0_variable_set_assignment" "parent" {
  scope    = "project"
  scope_id = var.project_id
  set_ids = [
    data.env0_variable_set.test1.id,
  ]
}

resource "env0_variable_set_assignment" "child" {
  scope    = "project"
  scope_id = var.child_project_id
  set_ids = [
    data.env0_variable_set.test2.id,
  ]
}

```

### PR DESCRIPTION
**Pull request by TomerHeber**
…able sets: 404 Not Found: Configuration Set "..." is not assigned to project.

### Issue & Steps to Reproduce / Feature Request
fixes #966 

### Solution

Modified the filter to check scopeId instead of scope.

### CODE REVIEW

File: env0/resource_variable_set_assignment_test.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func TestUnitVariableSetAssignmentResource(t *testing.T) {
 				AssignmentScope:   scope,
 				AssignmentScopeId: scopeId,
 			},
+			{
```
Comment (TomerHeber) [inline env0/resource_variable_set_assignment_test.go:64]: added this use case: this should be filtered out.
(Which mean the test outcome shouldn't change).

**Review by yaronya — APPROVED** [review]

**Comment by TomerHeber** [comment]
thank you @yaronya for the quick review (:

**Pull request merged**